### PR TITLE
SWATCH-5066: Update the IT Subscription Service UMB Consumer with logging

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: SWATCH-5066

## Description
Adding the following log statement:
```
log.info(
        "IT Subscription message consumed: source=umb, "
            + "subscriptionNumber={}, webCustomerId={}, sku={}, quantity={}, "
            + "effectiveStartDate={}, effectiveEndDate={}, terminated={}",
```            
that will allow us to compare the messages received from umb and kafka (future task).

Note that we don't need to add any specific metric because Quarkus already uses quarkus_messaging_message_count_total by channel that is the metric we need for grafana.

Refactored the UmbSubscription model to reuse the same logic in the service and the log. 

## Testing
As in https://github.com/RedHatInsights/rhsm-subscriptions/pull/5964, we don't need to update any tests. 

## Summary by CodeRabbit

* **Improvements**
  * More detailed UMB subscription logging showing subscription number, customer, resolved SKU, quantity, effective dates, and terminated-status presence.
  * UMB processing now respects the enabled flag and avoids emitting raw XML when disabled.
  * Safer subscription data handling: robust SKU and status lookups and more reliable determination of effective end dates.

* **Tests**
  * Added unit tests for subscription helper methods and edge-case scenarios.

### CODE REVIEW

File: swatch-contracts/src/main/java/com/redhat/swatch/contract/product/umb/UmbSubscription.java [hunk 1]
```diff
@@ -90,23 +91,28 @@ public String getWebCustomerId() {
...
+    return findProduct()
+        .map(SubscriptionProduct::getProduct)
+        .filter(Objects::nonNull)
+        .map(SubscriptionProduct::getStatus)
+        .filter(Objects::nonNull)
+        .flatMap(
+            statuses ->
+                Arrays.stream(statuses)
+                    .filter(status -> "Terminated".equalsIgnoreCase(status.getState()))
+                    .findFirst());
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply swatch-contracts/src/main/java/com/redhat/swatch/contract/product/umb/UmbSubscription.java:116]: updated

**Review by Mantzounis — APPROVED** [review]
lgtm!
